test/integration/smart_answer_flows/calculate_your_holiday_entitlement_test.rb

```diff
@@ -271,6 +271,274 @@ class CalculateYourHolidayEntitlementTest < ActiveSupport::TestCase
     end
   end # compressed hours
 
+  context "irregular hours" do
+    setup do
+      add_response "irregular-hours"
+    end
+    should "ask the time period for the calculation" do
+      assert_current_node :calculation_period?
+    end
+
+    context "answer full leave year" do
+      setup do
+        add_response "full-year"
+      end
+      should "calculate the holiday entitlement" do
+        SmartAnswer::Calculators::HolidayEntitlement
+          .expects(:new)
+          .with(
+            start_date: nil,
+            leaving_date: nil,
+            leave_year_start_date: nil
+          ).returns(@stubbed_calculator)
+        @stubbed_calculator.expects(:formatted_full_time_part_time_weeks).returns("5.6")
+
+        assert_state_variable "holiday_entitlement", "5.6"
+        assert_current_node :irregular_and_annualised_done
+      end
+    end
+
+    context "answer starting part way through the leave year" do
+      setup do
+        add_response "starting"
+      end
+      should "ask for the employment start date" do
+        assert_current_node :what_is_your_starting_date?
+      end
+      context "answer June 1st this year" do
+        setup do
+          add_response "#{Date.today.year}-06-01"
```

Comment:
Could the tests fail when the leap year (2020) comes round?

Reply:
This should be OK because we are stubbing the calculator to return the values we want, to test just that the flow communicates with the calculator as expected. The complicated leap-year/standard-year logic is all tested in the calculator itself.